tests/usm/usm_api.h

```diff
@@ -0,0 +1,688 @@
+ */
+template <typename T, size_t count, allocation sourceAllocation>
+class memcpy : public copyGeneric<T, count, sourceAllocation> {
+  using baseT = copyGeneric<T, count, sourceAllocation>;
+
+ public:
+  memcpy(sycl::queue &queue, int seed) : baseT(queue, seed) {}
+
+  /** @brief The memcpy() method supports non-USM pointers
+   */
+  static constexpr bool has_non_usm_support() { return true; }
+
+  template <allocation alloc>
+  static std::string description() {
+    return "memcpy from " + get_allocation_decription<sourceAllocation>() +
+           " to " + get_allocation_decription<alloc>();
+  }
+
+  template <typename parentT, typename... depEventsT>
+  auto call(parentT &parent, T *ptr, depEventsT &&... events) const {
+    constexpr auto size = sizeof(T) * count;
+    return parent.memcpy(ptr, this->source.get(), size,
+                         std::forward<depEventsT>(events)...);
+  }
+};
+}  // namespace detail
+
+template <typename T, size_t count>
+using copy_from_non_usm = detail::copy<T, count, allocation::non_usm>;
+template <typename T, size_t count>
+using copy_from_host = detail::copy<T, count, allocation::host>;
+template <typename T, size_t count>
+using copy_from_device = detail::copy<T, count, allocation::device>;
+template <typename T, size_t count>
+using copy_from_shared = detail::copy<T, count, allocation::shared>;
+
+template <typename T, size_t count>
+using memcpy_from_non_usm = detail::memcpy<T, count, allocation::non_usm>;
+template <typename T, size_t count>
+using memcpy_from_host = detail::memcpy<T, count, allocation::host>;
+template <typename T, size_t count>
+using memcpy_from_device = detail::memcpy<T, count, allocation::device>;
+template <typename T, size_t count>
+using memcpy_from_shared = detail::memcpy<T, count, allocation::shared>;
+
+/** @brief Provides test logic for the fill() method tests
+ */
+template <typename T, size_t count>
+class fill {
+  const T value;
+
+ public:
+  fill(sycl::queue &, int seed) : value{seed} {}
+
+  /** @brief The fill() method doesn't support non-USM pointers
+   */
+  static constexpr bool has_non_usm_support() { return false; }
+
+  template <allocation alloc>
+  static std::string description() {
+    return "fill using " + get_allocation_decription<alloc>();
+  }
+
+  template <typename parentT, typename... depEventsT>
+  auto call(parentT &parent, T *ptr, depEventsT &&... events) const {
+    return parent.fill(ptr, value, count, std::forward<depEventsT>(events)...);
+  }
+
+  void check(const T *ptr) const {
+    T reference[count];
+    std::fill_n(reference, count, value);
+
+    detail::check_values<count>(ptr, reference);
+  }
+};
+
+/** @brief Provides test logic for the memset() method tests
+ */
+template <typename T, size_t count>
+class memset {
+  const int value;
+
+ public:
+  static constexpr size_t size = count * sizeof(T);
+
+  memset(sycl::queue &, int seed) : value(seed) {}
+
+  /** @brief The memset() method doesn't support non-USM pointers
```

Comment:
```suggestion
  /** @brief The memset() member function doesn't support non-USM pointers
```

Reply:
I've replaced "method" to "member function" in this commit: 800816219c8f8b2a87887df4ef54c8088b0a153b